### PR TITLE
Add CI tests for Windows and Mac

### PR DESCRIPTION
**Pull request by JasonGross**

### CODE REVIEW

File: etc/bytedump.sh [hunk 1]
```diff
@@ -13,17 +13,18 @@ Abort.
...
-needle = b"\nCOQTOP_CRAP_ENDS_HERE\n"
+sep = os.linesep.encode("ascii")
+needle = b"COQTOP_CRAP_ENDS_HERE" + sep
 waiting_on_coqbug_15373 = True
 b = b""
 while True:
 	r = os.read(0, 1)
 	if not r:
-		sys.exit(waiting_on_coqbug_15373 + 2*(not b.endswith(b"\n")))
+		sys.exit(waiting_on_coqbug_15373 + 2*(not b.endswith(sep)))
```
Comment (JasonGross) [inline etc/bytedump.sh:23]: What about adding
```suggestion
		exitcode = waiting_on_coqbug_15373 + 2*(not b.endswith(sep))
		if exitcode:
		  print(f'Error: read {b!r}', file=sys.stderr)
		  sys.exit(exitcode)
```
for debugging purposes?

**Comment by JasonGross** [comment]
Seems shifted by 1
```
hexdump < /dev/null && \
	hexdump -C special/BytedumpTest.golden.bin > special/BytedumpTest.golden.hex && \
	hexdump -C special/BytedumpTest.out.tmp > special/BytedumpTest.out.hex && \
	diff -u special/BytedumpTest.golden.hex special/BytedumpTest.out.hex && \
	rm special/BytedumpTest.golden.hex special/BytedumpTest.out.hex || true
--- special/BytedumpTest.golden.hex	2022-09-29 12:51:05.396465200 +0000
+++ special/BytedumpTest.out.hex	2022-09-29 12:51:05.412087700 +0000
@@ -1,17 +1,18 @@
-00000000  00 01 02 03 04 05 06 07  08 09 0a 0b 0c 0d 0e 0f  |................|
-00000010  10 11 12 13 14 15 16 17  18 19 1a 1b 1c 1d 1e 1f  |................|
-00000020  20 21 22 23 24 25 26 27  28 29 2a 2b 2c 2d 2e 2f  | !"#$%&'()*+,-./|
-00000030  30 31 32 33 34 35 36 37  38 39 3a 3b 3c 3d 3e 3f  |0123456789:;<=>?|
-00000040  40 41 42 43 44 45 46 47  48 49 4a 4b 4c 4d 4e 4f  |@ABCDEFGHIJKLMNO|
-00000050  50 51 52 53 54 55 56 57  58 59 5a 5b 5c 5d 5e 5f  |PQRSTUVWXYZ[\]^_|
-00000060  60 61 62 63 64 65 66 67  68 69 6a 6b 6c 6d 6e 6f  |`abcdefghijklmno|
-00000070  70 71 72 73 74 75 76 77  78 79 7a 7b 7c 7d 7e 7f  |pqrstuvwxyz{|}~.|
-00000080  80 81 82 83 84 85 86 87  88 89 8a 8b 8c 8d 8e 8f  |................|
-00000090  90 91 92 93 94 95 96 97  98 99 9a 9b 9c 9d 9e 9f  |................|
-000000a0  a0 a1 a2 a3 a4 a5 a6 a7  a8 a9 aa ab ac ad ae af  |................|
-000000b0  b0 b1 b2 b3 b4 b5 b6 b7  b8 b9 ba bb bc bd be bf  |................|
-000000c0  c0 c1 c2 c3 c4 c5 c6 c7  c8 c9 ca cb cc cd ce cf  |................|
-000000d0  d0 d1 d2 d3 d4 d5 d6 d7  d8 d9 da db dc dd de df  |................|
-000000e0  e0 e1 e2 e3 e4 e5 e6 e7  e8 e9 ea eb ec ed ee ef  |................|
-000000f0  f0 f1 f2 f3 f4 f5 f6 f7  f8 f9 fa fb fc fd fe ff  |................|
-00000100
+00000000  00 01 02 03 04 05 06 07  08 09 0d 0a 0b 0c 0d 0e  |................|
+00000010  0f 10 11 12 13 14 15 16  17 18 19 1a 1b 1c 1d 1e  |................|
+00000020  1f 20 21 22 23 24 25 26  27 28 29 2a 2b 2c 2d 2e  |. !"#$%&'()*+,-.|
+00000030  2f 30 31 32 33 34 35 36  37 38 39 3a 3b 3c 3d 3e  |/0123456789:;<=>|
+00000040  3f 40 41 42 43 44 45 46  47 48 49 4a 4b 4c 4d 4e  |?@ABCDEFGHIJKLMN|
+00000050  4f 50 51 52 53 54 55 56  57 58 59 5a 5b 5c 5d 5e  |OPQRSTUVWXYZ[\]^|
+00000060  5f 60 61 62 63 64 65 66  67 68 69 6a 6b 6c 6d 6e  |_`abcdefghijklmn|
+00000070  6f 70 71 72 73 74 75 76  77 78 79 7a 7b 7c 7d 7e  |opqrstuvwxyz{|}~|
+00000080  7f 80 81 82 83 84 85 86  87 88 89 8a 8b 8c 8d 8e  |................|
+00000090  8f 90 91 92 93 94 95 96  97 98 99 9a 9b 9c 9d 9e  |................|
+000000a0  9f a0 a1 a2 a3 a4 a5 a6  a7 a8 a9 aa ab ac ad ae  |................|
+000000b0  af b0 b1 b2 b3 b4 b5 b6  b7 b8 b9 ba bb bc bd be  |................|
+000000c0  bf c0 c1 c2 c3 c4 c5 c6  c7 c8 c9 ca cb cc cd ce  |................|
+000000d0  cf d0 d1 d2 d3 d4 d5 d6  d7 d8 d9 da db dc dd de  |................|
+000000e0  df e0 e1 e2 e3 e4 e5 e6  e7 e8 e9 ea eb ec ed ee  |................|
+000000f0  ef f0 f1 f2 f3 f4 f5 f6  f7 f8 f9 fa fb fc fd fe  |................|
+00000100  ff                                                |.|
+00000101
```

**Comment by andres-erbsen** [comment]
Nah I think it has a CRLF in there :/.

I see https://github.com/mit-plv/bedrock2/actions/runs/3150591237/jobs/5123552583#step:9:1126 also failed -- did I just jump to conclusions that your last fix worked without actually seeing it work, or did I break it while removing the debug stuff...? Back to conference.

**Comment by JasonGross** [comment]
Old hex is `0a 0b 0c 0d 0e 0f`, new hex is `0d 0a 0b 0c 0d 0e 0f`.  This looks like a stray `\r`, not a CR LF.  I think the Python code is working fine, maybe Ltac2 is having an issue?

**Pull request merged**